### ISSUE TITLE
MatrixDepot does not initialize under julia1.4

### ISSUE THREAD

**Issue by KlausC** [JuliaMatrices/MatrixDepot.jl#40]
EDIT: reason for the following misbehavior was an obsolete cached data file `db.data`.
WORKAROUND: remove `MatrixDepot/data/db.data` file. It will be re-created when module is initialized next time.

Under julia1.4 the module is not usable.
Under 1.2 no problems,  status for julia1.3 is unknown.

```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.4.0-DEV.595 (2019-12-17)
 _/ |\__'_|_|_|\__'_|  |  krc/awmax/5fc8a05be4* (fork: 3 commits, 17 days)
|__/                   |

julia> using MatrixDepot
[ Info: Precompiling MatrixDepot [top-level]
include group.jl for user defined matrix generators
verify download of index files...
┌ Warning: exception during initialization: 'KeyError(MatrixDepot [b51810bb-c9f3-55da-ae3c-350fc1fbce05])'
└ @ MatrixDepot ~/.julia/dev/MatrixDepot/src/MatrixDepot.jl:135

julia> MatrixDepot.DATA_DIR
"/home/crusius/.julia/dev/MatrixDepot/data"

julia> MatrixDepot.mdinfo()
ERROR: MatrixDepot.DataError("no matrix according to baart found")

julia> MatrixDepot.init()
include group.jl for user defined matrix generators
verify download of index files...
ERROR: KeyError: key MatrixDepot [b51810bb-c9f3-55da-ae3c-350fc1fbce05] not found
Stacktrace:
 [1] getindex at ./dict.jl:477 [inlined]
 [2] root_module at ./loading.jl:960 [inlined]
 [3] deserialize_module(::Serialization.Serializer{IOStream}) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:912
 [4] handle_deserialize(::Serialization.Serializer{IOStream}, ::Int32) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:812
 [5] deserialize(::Serialization.Serializer{IOStream}) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:735
 [6] deserialize_datatype(::Serialization.Serializer{IOStream}, ::Bool) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:1210
 [7] handle_deserialize(::Serialization.Serializer{IOStream}, ::Int32) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:788
 [8] deserialize(::Serialization.Serializer{IOStream}) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:735
 [9] handle_deserialize(::Serialization.Serializer{IOStream}, ::Int32) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:795
 [10] deserialize(::Serialization.Serializer{IOStream}) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:735
 [11] handle_deserialize(::Serialization.Serializer{IOStream}, ::Int32) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:838
 [12] deserialize(::Serialization.Serializer{IOStream}) at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:735
 [13] deserialize at /home/julia/julia/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:722 [inlined]
 [14] (::MatrixDepot.var"#100#101"{MatrixDepot.MatrixDatabase})(::IOStream) at /home/crusius/.julia/dev/MatrixDepot/src/download.jl:101
 [15] open(::MatrixDepot.var"#100#101"{MatrixDepot.MatrixDatabase}, ::String, ::Vararg{String,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:298
 [16] open at ./io.jl:296 [inlined]
 [17] readdb at /home/crusius/.julia/dev/MatrixDepot/src/download.jl:100 [inlined]
 [18] downloadindices(::MatrixDepot.MatrixDatabase; ignoredb::Bool) at /home/crusius/.julia/dev/MatrixDepot/src/download.jl:125
 [19] init(; ignoredb::Bool) at /home/crusius/.julia/dev/MatrixDepot/src/MatrixDepot.jl:128
 [20] init() at /home/crusius/.julia/dev/MatrixDepot/src/MatrixDepot.jl:102
 [21] top-level scope at REPL[4]:1

julia> 
```

**Comment by KlausC** [comment]
Maybe related to PR #39 .

**Comment by KlausC** [comment]
Resolved in v1.0
